tests/draft4/optional/format/hostname.json

```diff
@@ -92,6 +92,16 @@
                 "description": "exceeds maximum label length",
                 "data": "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijkl.com",
                 "valid": false
+            },
+            {
+                "description": "single label with hyphen",
+                "data": "host-name",
+                "valid": true
+            },
+            {
+                "description": "single label starting with digit",
+                "data": "1host",
```

Comment:
This test isn't correct for earlier drafts (4, 6, and 7), which pointed to RFC1034 (which disallows leading digits). For newer ones we point to RFC1123.

Reply:
I changed the validity to `false` for the drafts referencing RFC1034.
Is this okay, or shall I remove the tests completely?

Reply:
Changing them seems fine to me too.

> I don't think that anyone is really expecting those domains to fail validation even on an older version.

I'm not sure I understand what you mean to point out here -- I agree if you're trying to say people might not know this, but it doesn't matter much what people expect, what matters is what the spec says.

Reply:
> but it doesn't matter much what people expect, what matters is what the spec says.

I disagree. This was a mistake in the older version of the json schema spec, hostnames with leading digits are used for a long time in the wild.

What matters in the end is the software behavior is expected and if it breaks or not in deployments.
The spec is just a way to communicate between developers of different software parts to ensure that the behavior is expected, it is a tool, not the terminal goal.

Enforcing e.g. draft-07 impls to follow outdated ineffective rfc that will break existing websites is counter-productive.

Better to remove these at all from 4/6/7, I think.

Refs: https://github.com/json-schema-org/json-schema-spec/pull/716
Refs: https://webmasters.stackexchange.com/a/814 (very old, currently there are more examples)

Reply:
The test suite is not the venue to make those judgements about what is right or wrong when the specifiation says something.

That being said (though it seems I wrote "changing" above when I meant removing), I'm ok with removing them too -- @swaeberle let's go with dropping those tests from these versions.

Reply:
Okay, I dropped those tests from the older drafts.